Sources/Subs-Db-mysql.php

```diff
@@ -920,4 +921,48 @@ function smf_is_resource($result)
 	return false;
 }
 
+/**
+ * Updates data in a table, using data from other tables
+ *
+ * @param array $table Associative array with info about the table to be updated ('name' => '{db_prefix}foo', 'alias' => 'f')
+ * @param array $from Array of associative arrays with info about the other tables to get data from ('name' => '{db_prefix}bar', 'alias' => 'b', 'condition' => 'f.baz = b.qux')
+ * @param string $set A string containing the SET instructions for the update query
+ * @param string $where A string containing any WHERE conditions for the update query
+ * @param array $db_values The values to be inserted into the compiled query string
+ * @param object $connection The connection to use (if null, $db_connection is used)
+ * @return bool True if the update was successful, otherwise false
+ */
+function smf_db_update_from($table, $joined, $set, $where, $db_values, $connection = null)
+{
+	global $smcFunc, $db_connection, $db_prefix;
+
+	$connection = $connection === null ? $db_connection : $connection;
+
+	if (empty($table['name']) || empty($table['alias']) || empty($set))
+		return;
+
+	$joins = array();
+	foreach ($joined as $join)
+	{
+		if (empty($join['name']) || empty($join['alias']) || empty($join['condition']))
+			continue;
+
+		$joins[] = 'INNER JOIN ' . $join['name'] . ' AS ' . $join['alias'] . ' ON (' . $join['condition'] . ')';
```

Comment:
please let the "inner" away this works fine without this word.
the inner thing is similiar to "the dead person is dead" (we know that he is dead we don't need call it twice.)
and the same with inner, we know that a inner because there is no left or right.